uk.co.compucorp.civicrm.tasksassignments/js/test/tasks-assignments/controllers/modal/modal-assignment.controller.spec.js

```diff
@@ -71,38 +71,60 @@ define([
     });
 
     describe('confirm()', function () {
+      var expectedDocumentsParameters, expectedTasksParameters;
+      var expectedCaseId = _.uniqueId();
+      var expectedContactId = _.uniqueId();
+
       beforeEach(function () {
         spyOn($q, 'all').and.returnValue({ then: _.noop });
         spyOn(taskService, 'saveMultiple');
         spyOn(documentService, 'saveMultiple');
         spyOn(assignmentService, 'save').and.returnValue({
           then: function (callback) {
             // fakes the db permanence, attaching an id to the assignment on the scope
-            callback(_.assign(_.clone(scope.assignment), { id: _.uniqueId() }));
+            callback(_.assign(_.clone(scope.assignment), { id: expectedCaseId }));
           }
         });
       });
+
       beforeEach(function () {
         prepAssignment();
-        scope.confirm() && scope.$digest();
-      });
 
-      it('does not pass the original documents objects to the documentService', function () {
-        var documents = documentService.saveMultiple.calls.mostRecent().args[0];
+        expectedDocumentsParameters = getActivitiesToBeSentToAPI(scope.documentList);
+        expectedTasksParameters = getActivitiesToBeSentToAPI(scope.taskList);
 
-        expect(documents.every(function (doc, index) {
-          return doc !== scope.taskList[index];
-        })).toBe(true);
+        scope.confirm() && scope.$digest();
       });
 
-      it('does not pass the original tasks objects to the taskService', function () {
-        var tasks = taskService.saveMultiple.calls.mostRecent().args[0];
+      it('saves every document', function () {
+        expect(documentService.saveMultiple).toHaveBeenCalledWith(expectedDocumentsParameters);
+      });
 
-        expect(tasks.every(function (task, index) {
-          return task !== scope.taskList[index];
-        })).toBe(true);
+      it('saves every task', function () {
+        expect(taskService.saveMultiple).toHaveBeenCalledWith(expectedTasksParameters);
       });
 
+      /**
+       * Returns a list of activities containing the fields required by their
+       * respective services. It will only return activities marked for creation.
+       *
+       * @param {Array} activities a list of activities that are missing their assignee and case ids.
```

Comment:
`* @param  {Array} activities a list of activities missing their assignee and case IDs.`

just a bit more succinct version.